### ISSUE TITLE
curation store is not queryable because of unescaped `.` characters

### ISSUE THREAD

**Issue by dabutvin** [clearlydefined/service#411]
We have been able to insert the data, but we are not able to query it back out.

see https://jira.mongodb.org/browse/SERVER-30575
you can insert the data, but you cannot query it

Existing structure below.
for example, this query returns no results
```
find( ({'files.curations/nuget/nuget/-/Microsoft.Extensions.Localization.Abstractions.yaml': { $exists: true })
```

```json
{
    "_id" : 781,
    "pr" : {
        "number" : 781,
        "id" : 233651751,
        "state" : "closed",
        "title" : "ariel11_181126_181741.275",
        "user" : {
            "login" : "ariel11"
        },
        "body" : "\n**Type:** Missing\n\n**Summary:**\nDeclaring\n\n**Details:**\nDeclaring License \n\n**Resolution:**\nDeclaring License for v2.0.2 https://github.com/aspnet/AspNetCore/blob/c4c6e1b0e9667662646939139205e12bf57e865c/LICENSE.txt\n\n**Affected definitions**:\n- Microsoft.Extensions.Localization.Abstractions 2.0.2,- Microsoft.Extensions.Options 2.0.2",
        "created_at" : "2018-11-26T18:17:44Z",
        "updated_at" : "2018-11-27T22:59:28Z",
        "closed_at" : "2018-11-27T22:59:24Z",
        "merged_at" : "2018-11-27T22:59:24Z",
        "merge_commit_sha" : "278e11c66e1eb07a240b83808385114d57e01c50",
        "head" : {
            "sha" : "b506af427679972cc3406265236c61889c61441f",
            "repo" : {
                "id" : 113909603
            }
        },
        "base" : {
            "sha" : "77cc5b0bc76116a0b2f0a88bf16f8ca9cc7c4ffe",
            "repo" : {
                "id" : 113909603
            }
        }
    },
    "files" : {
        "curations/nuget/nuget/-/Microsoft.Extensions.Localization.Abstractions.yaml" : {
            "coordinates" : {
                "name" : "Microsoft.Extensions.Localization.Abstractions",
                "provider" : "nuget",
                "type" : "nuget"
            },
            "revisions" : {
                "2.0.0" : {
                    "licensed" : {
                        "declared" : "Apache-2.0"
                    }
                },
                "2.0.1" : {
                    "licensed" : {
                        "declared" : "Apache-2.0"
                    }
                },
                "2.0.2" : {
                    "licensed" : {
                        "declared" : "Apache-2.0"
                    }
                },
                "2.1.0" : {
                    "licensed" : {
                        "declared" : "Apache-2.0"
                    }
                },
                "2.1.1" : {
                    "licensed" : {
                        "declared" : "Apache-2.0"
                    }
                }
            }
        }
    }
}
```

**Comment by dabutvin** [comment]
proposed new structure. Little more verbose

```json
{
  "_id": 781,
  "pr": {
    "number": 781,
    "id": 233651751,
    "state": "closed",
    "title": "ariel11_181126_181741.275",
    "user": {
      "login": "ariel11"
    },
    "body": "\n**Type:** Missing\n\n**Summary:**\nDeclaring\n\n**Details:**\nDeclaring License \n\n**Resolution:**\nDeclaring License for v2.0.2 https://github.com/aspnet/AspNetCore/blob/c4c6e1b0e9667662646939139205e12bf57e865c/LICENSE.txt\n\n**Affected definitions**:\n- Microsoft.Extensions.Localization.Abstractions 2.0.2,- Microsoft.Extensions.Options 2.0.2",
    "created_at": "2018-11-26T18:17:44Z",
    "updated_at": "2018-11-27T22:59:28Z",
    "closed_at": "2018-11-27T22:59:24Z",
    "merged_at": "2018-11-27T22:59:24Z",
    "merge_commit_sha": "278e11c66e1eb07a240b83808385114d57e01c50",
    "head": {
      "sha": "b506af427679972cc3406265236c61889c61441f",
      "repo": {
        "id": 113909603
      }
    },
    "base": {
      "sha": "77cc5b0bc76116a0b2f0a88bf16f8ca9cc7c4ffe",
      "repo": {
        "id": 113909603
      }
    }
  },
  "files": [
    {
      "path": "curations/nuget/nuget/-/Microsoft.Extensions.Localization.Abstractions.yaml",
      "coordinates": {
        "name": "Microsoft.Extensions.Localization.Abstractions",
        "provider": "nuget",
        "type": "nuget"
      },
      "revisions": [
        {
          "revision": "2.0.0",
          "data": {
            "licensed": {
              "declared": "Apache-2.0"
            }
          }
        },
        {
          "revision": "2.0.1",
          "data": {
            "licensed": {
              "declared": "Apache-2.0"
            }
          }
        },
        {
          "revision": "2.0.2",
          "data": {
            "licensed": {
              "declared": "Apache-2.0"
            }
          }
        },
        {
          "revision": "2.1.0",
          "data": {
            "licensed": {
              "declared": "Apache-2.0"
            }
          }
        },
        {
          "revision": "2.1.1",
          "data": {
            "licensed": {
              "declared": "Apache-2.0"
            }
          }
        }
      ]
    }
  ]
}

```

we can do the queries like

```js
db.collection.find({
    'files.coordinates.type': 'nuget',
    'files.coordinates.provider': 'nuget',
    'files.coordinates.namespace': null,
    'files.coordinates.name': 'Microsoft.Extensions.Localization.Abstractions',
    'files.revisions.revision': '2.0.1'
 })
```

and 

```js
db.collection.find({
    'files.revisions.data.licensed.declared': 'Apache-2.0'
 })
```

**Comment by jeffmcaffer** [comment]
Looks good

**Comment by dabutvin** [comment]
fixed in #412
